src/components/cloud-media/index.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { ImgHTMLAttributes } from 'react';
+import classNames from 'classnames';
+import { AdvancedImage, AdvancedVideo } from '@cloudinary/react';
+import { CloudinaryVideo } from '@cloudinary/url-gen';
+import { pad } from '@cloudinary/url-gen/actions/resize';
+import { auto } from '@cloudinary/url-gen/qualifiers/background';
+import { getCloudMedia, getHashFromIpfsUrl } from './utils';
+import { CloudinaryMedia, MediaType } from './types';
+
+export interface ComponentProperties
+  extends ImgHTMLAttributes<HTMLImageElement> {
+  className: string;
+  getCloudMedia: (
+    publicId: string,
+    prefix?: string
+  ) => Promise<CloudinaryMedia>;
+  getHashFromIpfsUrl: (url: string) => string;
+}
+
+interface State {
+  isLoaded: boolean;
+  cloudinaryMedia: CloudinaryMedia;
+  isLoading: boolean;
+}
+
+export class Component extends React.Component<ComponentProperties, State> {
+  state = { isLoaded: false, cloudinaryMedia: null, isLoading: true };
+
+  async componentDidMount() {
+    if (this.props.src) {
+      await this.fetchMedia();
```

Comment:
this is minor, but you shouldn't need to await, this. it's minor because it doesn't actually block the lifecycle, but i think it just adds visual clutter, and could possibly lead to confusion. even though `fetchMedia` is async, you can just call it here.